ui/src/app/applications/components/application-summary/application-summary.tsx

```diff
@@ -278,21 +280,25 @@ export const ApplicationSummary = (props: {app: models.Application; updateApp: (
     async function setAutoSync(ctx: {popup: PopupApi}, confirmationTitle: string, confirmationText: string, prune: boolean, selfHeal: boolean) {
         const confirmed = await ctx.popup.confirm(confirmationTitle, confirmationText);
         if (confirmed) {
+            setChangeSync(true);
             const updatedApp = JSON.parse(JSON.stringify(props.app)) as models.Application;
             if (!updatedApp.spec.syncPolicy) {
                 updatedApp.spec.syncPolicy = {};
             }
             updatedApp.spec.syncPolicy.automated = {prune, selfHeal};
-            props.updateApp(updatedApp);
+            await props.updateApp(updatedApp);
+            setChangeSync(false);
         }
     }
 
     async function unsetAutoSync(ctx: {popup: PopupApi}) {
         const confirmed = await ctx.popup.confirm('Disable Auto-Sync?', 'Are you sure you want to disable automated application synchronization');
         if (confirmed) {
+            setChangeSync(true);
             const updatedApp = JSON.parse(JSON.stringify(props.app)) as models.Application;
             updatedApp.spec.syncPolicy.automated = null;
-            props.updateApp(updatedApp);
```

Comment:
Please move `setChangeSync(false);` into final block to make sure spinner is removed even if `props.updateApp(updatedApp)` fails

Reply:
Hi Alex, done.